src/librustc/session/search_paths.rs

```diff
@@ -32,12 +28,16 @@ pub enum PathKind {
     All,
 }
 
-impl SearchPaths {
-    pub fn new() -> SearchPaths {
-        SearchPaths { paths: Vec::new() }
+impl PathKind {
+    pub fn matches(kind1: PathKind, kind2: PathKind) -> bool {
+        kind1 == PathKind::All ||
+        kind2 == PathKind::All ||
+        kind1 == kind2
```

Comment:
I think this would be more readable like this:
```rust
match (kind1, kind2) {
    (PathKind::All, _) | (_, PathKind::All) => true,
    _ => kind1 == kind2,
}
```